src/pyscipopt/scip.pyx

```diff
@@ -4167,11 +4171,7 @@ cdef class Model:
         """
         if sol == None:
             sol = Solution.create(self._scip, NULL)
-        if isinstance(expr, Variable):
-            var = <Variable> expr
-            return SCIPgetSolVal(self._scip, sol.sol, var.scip_var)
-        else:
-            return expr._evaluate(sol)
+        return sol[expr]
```

Comment:
So here. before there was this code checking whether expr is a variable. What I would argue is that if expr is a variable and sol is None, then we call directly `SCIPgetSolVal(self._scip, NULL, var.scip_var)` or `SCIPvarGetSol`. My point is that we do not need to create a wrapper for the NULL solution in such a case

Reply:
actually, we could even have a nullsol stored in model and we always use that one. @mattmilten @CGraczyk ?